package/cloudshell/iac/terraform/models/tf_service.py

```diff
@@ -0,0 +1,36 @@
+import uuid
+
+from cloudshell.api.cloudshell_api import CloudShellAPISession, AttributeNameValue
+
+from cloudshell.iac.terraform.constants import ATTRIBUTE_NAMES
+
+
+class TerraformServiceObject(object):
+    def __init__(self, api: CloudShellAPISession, res_id: str, name: str, cloudshell_model_name: str):
+        self.api = api
+        self.res_id = res_id
+        self.name = name
+        self.cloudshell_model_name = cloudshell_model_name
+        self.attributes = self._set_context_resource_attributes()
+        self._check_uuid()
+
+    def _check_uuid(self):
+        if f"{self.cloudshell_model_name}.{ATTRIBUTE_NAMES.UUID}" not in self.attributes:
+            raise ValueError(f"{ATTRIBUTE_NAMES.UUID} attribute was not found")
+        if not self.attributes[f"{self.cloudshell_model_name}.{ATTRIBUTE_NAMES.UUID}"]:
+            new_uuid = uuid.uuid4().hex
+            self.attributes[ATTRIBUTE_NAMES] = new_uuid
```

Comment:
it should be "self.attributes[ATTRIBUTE_NAMES.UUID] = new_uuid" ?

Reply:
Amazing catch...